andela/ah-magnificent6
# #159965317 Add search and filter to articles

**What does this PR do?**
Adds search and filter functionality ro articles

**Description of Task to be completed?**
Add search and filter to endpoint `GET api/articles` by:
- Author
- Title
- Tags

**How should this be manually tested?**
Run `git clone https://github.com/andela/ah-magnificent6.git` from your local machine to clone the repo. 
Run `ft-add-search-functionality-159965317` to go to the relevant branch. 
Run `python3 -m venv venv` and `source venv/bin/activate` to activate it. 
Run `pip install -r requirements.txt` to install the required packages.
Create a `.env` file from the `.env-sample` and run `python manage.py runserver` to start the app.
Go to the link [http://127.0.0.1:8000](http://127.0.0.1:8000) to view the documentation and test it
Register 2 users through the endpoint `POST /api/user/signup`
The expected result should have a token
Authorize using that token in the format `Bearer <token>`
Access protected endpoint `POST /api/articles/`
Post articles with the different users with different tags and titles
Search using`GET /api/articles/?search=search_term`
Filter using `GET /api/articles/?author__username__icontains=author&article_tags__tag__icontains=tag&tile__icontain=string in article`
It should filter using the values you input

**What are the relevant pivotal tracker stories?**
[#159965317](https://www.pivotaltracker.com/story/show/159965317)
